app/containers/App/reducer.js

```diff
@@ -10,42 +10,73 @@
  *   return state.set('yourStateVariable', true);
  */
 
+import { combineReducers } from 'redux-immutable';
+
 import {
   LOAD_REPOS_SUCCESS,
   LOAD_REPOS,
   LOAD_REPOS_ERROR,
 } from './constants';
-import { fromJS } from 'immutable';
-
-// The initial state of the App
-const initialState = fromJS({
-  loading: false,
-  error: false,
-  currentUser: false,
-  userData: {
-    repositories: false,
-  },
-});
+import { List } from 'immutable';
+
+function loading(state = false, action) {
+  switch (action.type) {
+    case LOAD_REPOS:
+      return true;
+    case LOAD_REPOS_SUCCESS:
+      return false;
+    case LOAD_REPOS_ERROR:
+      return false;
+    default:
+      return state;
+  }
+}
 
-function appReducer(state = initialState, action) {
+function error(state = false, action) {
   switch (action.type) {
     case LOAD_REPOS:
-      return state
-        .set('loading', true)
-        .set('error', false)
-        .setIn(['userData', 'repositories'], false);
+      return false;
     case LOAD_REPOS_SUCCESS:
-      return state
-        .setIn(['userData', 'repositories'], action.repos)
-        .set('loading', false)
-        .set('currentUser', action.username);
+      return false;
     case LOAD_REPOS_ERROR:
-      return state
-        .set('error', action.error)
-        .set('loading', false);
+      return action.error;
     default:
       return state;
   }
 }
 
-export default appReducer;
+function currentUser(state = false, action) {
+  switch (action.type) {
+    case LOAD_REPOS_SUCCESS:
+      return action.username;
+    default:
+      return state;
+  }
+}
+
+function repositories(state = false, action) {
+  if (List.isList(state)) {
+    // normalize the state from the server
+    state = state.toJS(); // eslint-disable-line no-param-reassign
+  }
+
+  switch (action.type) {
+    case LOAD_REPOS:
+      return false;
+    case LOAD_REPOS_SUCCESS:
+      return action.repos;
+    default:
+      return state;
+  }
+}
+
+const userData = combineReducers({
+  repositories,
+});
+
+export default combineReducers({
+  loading,
+  error,
+  currentUser,
+  userData,
+});
```

Comment:
I think It's already hard and complicated to edit 3 different files (actions.js, constants.js, reducer.js) and keep track of importing and exporting to have a reducer working compared to easy-to-reason [compact code](https://github.com/infinitered/ignite/blob/master/ignite-base/App/Redux/TemperatureRedux.js) with [modularity](https://github.com/erikras/ducks-modular-redux).